src/main/java/com/fasterxml/jackson/databind/deser/DeserializationProblemHandler.java

```diff
@@ -235,7 +235,43 @@ public Object handleUnexpectedToken(DeserializationContext ctxt,
     {
         return NOT_HANDLED;
     }
-    
+
+    /**
+     * Method that deserializers should call if the first token of the value to
+     * deserialize is of unexpected type (that is, type of token that deserializer
+     * cannot handle). This could occur, for example, if a Number deserializer
+     * encounter {@link JsonToken#START_ARRAY} instead of
+     * {@link JsonToken#VALUE_NUMBER_INT} or {@link JsonToken#VALUE_NUMBER_FLOAT}.
+     *<ul>
+     * <li>Indicate it does not know what to do by returning {@link #NOT_HANDLED}
+     *  </li>
+     * <li>Throw a {@link IOException} to indicate specific fail message (instead of
+     *    standard exception caller would throw
+     *  </li>
+     * <li>Handle content to match (by consuming or skipping it), and return actual
+     *    instantiated value (of type <code>targetType</code>) to use as replacement;
+     *    value may be `null` as well as expected target type.
+     *  </li>
+     * </ul>
+     *
+     * @param failureMsg Message that will be used by caller
+     *    to indicate type of failure unless handler produces value to use
+     *
+     * @return Either {@link #NOT_HANDLED} to indicate that handler does not know
+     *    what to do (and exception may be thrown), or value to use (possibly
+     *    <code>null</code>
+     *
+     * @since 2.8
```

Comment:
Would need to be 2.10 (if added in 2.10)

Reply:
Also, method being forwarded to should be deprecated; and all call points to it retrofitted to call the new method (which I think you are doing)

Reply:
I've deprecated the old method and changed `since` value. Please take a look.